src/content_manager.cc

```diff
@@ -984,8 +985,13 @@ int ContentManager::addContainerChain(const std::string& chain, const std::strin
     if (chain.empty())
         throw_std_runtime_error("addContainerChain() called with empty chain parameter");
 
-    log_debug("received chain: {} ({}) [{}]", chain.c_str(), lastClass.c_str(), dict_encode_simple(lastMetadata).c_str());
-    storage->addContainerChain(chain, lastClass, lastRefID, &containerID, &updateID, lastMetadata);
+    std::string newChain = chain;
+    for (const auto& pattern : config->getDictionaryOption(CFG_IMPORT_LAYOUT_MAPPING)) {
+        newChain = std::regex_replace(newChain, std::regex(pattern.first), pattern.second);
```

Comment:
I'm concerned about 2 things:
1. It's a regex. Docs don't mention it so it could become a surprise if used with square brackets or point. Regexes are easy to get wrong.
2. This affects also js layout. Probably not what user expects (however quite unlikely to both use js layout and have mapping...).

Reply:
1. is  good point. Added info
2. Very likely if you have separate media source. Easier than adding all maps in your script

Reply:
> Easier than adding all maps in your script

That is just the same regex replace, no?

But as I said, it's likely won't be a problem in practice.